src/lib/api/basyx-v3/api.ts

```diff
@@ -540,6 +653,41 @@ export const SubmodelRepositoryApiFp = function (configuration?: Configuration)
                 }
             };
         },
+
+        /**
+         * @summary Uploads an attachment to a specific submodel element.
+         * @param {string} submodelId - The unique identifier of the submodel containing the submodel element.
+         * @param {AttachmentDetails} attachmentDetails - The attachment data to be uploaded to the submodel element.
+         * @param {object} [options] - Optional. Additional options to override default HTTP request settings.
+         * @throws {RequiredError}
+         */
+        putAttachmentToSubmodelElement(submodelId: string, attachmentDetails: AttachmentDetails, options: any) {
+            return async (requestHandler: FetchAPI = isomorphicFetch, basePath: string = BASE_PATH) => {
+                const localVarRequestOptions = Object.assign({ method: 'PUT' }, options);
```

Comment:
Same as above, I think it's strange that the function returns a callback instead of performing the operation directly.